packages/core/lib/probability-manager.ts

```diff
@@ -102,6 +112,13 @@ class ProbabilityManager {
         if (lastProbabilityTimeBeforeTimeout === this.lastProbabilityTime) {
           this.setProbability(probability)
         }
+
+        // Initial sampling request has been made, and we can unblock batching
+        if (milliseconds === 0 && this.resolveInitialProbability) {
```

Comment:
The `milliseconds === 0` isn't doing much work, and could confuse things further down the road. I think it's probably safe to remove:

```suggestion
        if (this.resolveInitialProbability) {
```